### ISSUE TITLE
Layout/ClassStructure: doesn't properly recognize class methods on Ruby 2.3.x 

### ISSUE THREAD

**Issue by jbodah** [rubocop-hq/rubocop#6491]
`Layout/ClassStructure` doesn't recognize `class_methods` nodes on Ruby 2.3.x. It can be fixed by explicitly adding `class_methods` to the `ExpectedOrder` (note adding `class_methods` under `Categories` does _not_ work though):

```
Layout/ClassStructure:
  Enabled: true
  ExpectedOrder:
    - class_methods
    - initializer
    - public_methods
    - protected_methods
    - private_methods
```

--------

## Expected behavior

The repro example below should fail due to class method below instance methods

## Actual behavior

Example passes without warning/failure

## Steps to reproduce the problem

```rb
class Hello
  CONST = 123

  def initialize
    @name = "hello"
  end

  def do_the_thing
    "blah"
  end

  def self.do_the_other_thing
    "blah"
  end
end
```

The cop was patched to print the AST node (if not ignored) as well as the classification. You can see that `class_method` is ignored by default

```
$ rubocop hello.rb --only Layout/ClassStructure
Inspecting 1 file
constants
(casgn nil :CONST
  (int 123))
constants
initializer
(def :initialize
  (args)
  (ivasgn :@name
    (str "hello")))
initializer
public_methods
(def :do_the_thing
  (args)
  (str "blah"))
public_methods
class_methods
.

1 file inspected, no offenses detected
```

## RuboCop version

```rb
$ ruby --version
ruby 2.3.2p217 (2016-11-15 revision 56796) [x86_64-darwin17]
$ rubocop --version
0.59.2
```

```
$ [bundle exec] rubocop -V
0.60.0 (using Parser 2.5.1.2, running on ruby 2.5.1 x86_64-linux)
```

**Comment by jbodah** [comment]
Ruby 2.3 is EOL; closing

**Comment by golubitsky** [comment]
This is still an issue as of

```
$ ruby -v
ruby 2.6.5p114 (2019-10-01 revision 67812) [x86_64-darwin18]
$ rubocop -v
1.17.0
```

test.rb

```
# frozen_string_literal: true

class Hello
  def instance_method
    # pass
  end

  def self.class_method
    # pass
  end
end
```

.rubocop.yml

```
Layout/ClassStructure:
  Enabled: true
  ExpectedOrder:
    - public_class_methods
    - public_methods
```

### Expected behavior
An offense on `Layout/ClassStructure`  

### Actual behavior

```
$ rubocop test.rb
Inspecting 1 file
.

1 file inspected, no offenses detected
```
